internal/provider/host_resource_retry.go

```diff
@@ -0,0 +1,116 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+// HostOperationFunc defines the signature for operations that can be retried
+type HostOperationFunc func() ([]byte, diag.Diagnostics, int)
+
+// HostRetryConfig contains configuration for retrying host operations
+type HostRetryConfig struct {
+	stateConf          *retry.StateChangeConf
+	operationName      string
+	operation          HostOperationFunc
+	successStatusCodes []int
+	ctx                context.Context
+}
+
+// Retry state constants
+const (
+	hostRetryStateRetrying = "retrying"
+	hostRetryStateSuccess  = "success"
+)
+
+// Default Retry Time Constants
+const (
+	defaultRetryTimeout     = 1800 // Overall timeout for retry (seconds) Default: 30min
+	defaultRetryDelay       = 5    // Time to wait between retries (seconds)
+	defaultRetryInitalDelay = 2    // Initial delay before first retry (seconds)
+)
+
+// CreateRetryConfig creates a StateChangeConf for retrying operations with exponential backoff.
+// This follows Terraform provider best practices for handling transient API errors.
+//
+// Common retryable scenarios based on RFC 7231 and industry standards:
+// - HTTP 409: Resource conflict (host in use by running jobs)
+// - HTTP 408: Request timeout
+// - HTTP 429: Too many requests (rate limiting)
+// - HTTP 500: Internal server error
+// - HTTP 502: Bad gateway
+// - HTTP 503: Service unavailable
+// - HTTP 504: Gateway timeout
+//
+// Uses the provided timeout seconds instead of calculating from context deadline.
+func CreateRetryConfig(operationName string, operation HostOperationFunc, successStatusCodes []int,
+	retryableStatusCodes []int, retryTimeout int64, initialDelay int64, retryDelay int64,
+) *HostRetryConfig {
+	// Use provided delays, fallback to defaults if zero
+	if retryTimeout == 0 {
+		retryTimeout = defaultRetryTimeout
+	}
+	if initialDelay == 0 {
+		initialDelay = defaultRetryDelay
+	}
+	if retryDelay == 0 {
+		retryDelay = defaultRetryInitalDelay
+	}
```

Comment:
Seems like a second default implementation, lets remove this 0 checking behaviour 😄